src/renderer/api/kube-api.ts

```diff
@@ -127,15 +127,15 @@ export function ensureObjectSelfLink(api: KubeApi, object: KubeJsonApiData) {
   }
 }
 
-export type KubeApiWatchCallback = (data: IKubeWatchEvent, error: any) => void;
+export type KubeApiWatchCallback = (data: IKubeWatchEvent<KubeJsonApiData>, error: any) => void;
 
 export type KubeApiWatchOptions = {
   namespace: string;
   callback?: KubeApiWatchCallback;
   abortController?: AbortController
 };
 
-export class KubeApi<T extends KubeObject = any> {
+export class KubeApi<T extends KubeObject> {
```

Comment:
Yes that would work. I can change it back.

Reply:
It is a breaking change yes, though only if you take it as an argument, you don't need to specify it when doing `new KubeApi(...)`.

Furthermore, from a TS perspective it is better not to have this as a default parameter because then the passed in `objectConstructor` doesn't get to "set" this type parameter.

Reply:
Please describe this in the PR description (maybe PR title also needs an update?). I also think we need to add `area/extension` label because this affects extension API.